Jolt/Physics/Collision/Shape/PlaneShape.h

```diff
@@ -0,0 +1,143 @@
+// Jolt Physics Library (https://github.com/jrouwe/JoltPhysics)
+// SPDX-FileCopyrightText: 2024 Jorrit Rouwe
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <Jolt/Physics/Collision/Shape/Shape.h>
+#include <Jolt/Physics/Collision/Shape/SubShapeID.h>
+#include <Jolt/Physics/Collision/PhysicsMaterial.h>
+
+JPH_NAMESPACE_BEGIN
+
+class CollideShapeSettings;
+
+/// Class that constructs a PlaneShape
+class JPH_EXPORT PlaneShapeSettings final : public ShapeSettings
+{
+public:
+	JPH_DECLARE_SERIALIZABLE_VIRTUAL(JPH_EXPORT, PlaneShapeSettings)
+
+	/// Default constructor for deserialization
+									PlaneShapeSettings() = default;
+
+	/// Create a plane shape.
+									PlaneShapeSettings(const Plane &inPlane, const PhysicsMaterial *inMaterial = nullptr, float inSize = cDefaultSize) : mPlane(inPlane), mMaterial(inMaterial), mSize(inSize) { }
+
+	// See: ShapeSettings
+	virtual ShapeResult				Create() const override;
+
+	Plane							mPlane;														///< Plane that describes the shape. The negative half space is considered solid.
+
+	RefConst<PhysicsMaterial>		mMaterial;													///< Surface material of the plane
+
+	static constexpr float			cDefaultSize = 1000.0f;										///< Default size of the plane
+
+	float							mSize = cDefaultSize;										///< The bounding box of this plane will run from [-size, size]. Keep this as low as possible for better broad phase performance.
```

Comment:
Somewhat nitpicky, and I realize this is clarified in the comment here, but given that `BoxShape` calls its bounds `half extent` I can't help but feel like maybe this should be referred to as "half size" rather than "size".

The fact that the `GetSize()` says that it's the "size of the bounding box" adds to the confusion a bit as well, since it's technically half the size of the bounding box (along the two perpendicular vectors).

Reply:
You're right. I've renamed mSize to mHalfExtent.

W.r.t. AABox, it would have been better to name that GetHalfExtent too, but I'm trying to keep the API stable so I'll leave that as is.